### PR TITLE
feat: Add import / export settings button

### PR DESCRIPTION
**Pull request by Larsluph**
# Add import / export settings button [feat]

Add buttons to import / export settings configuration, UI/UX design might be reworked 😅

image

# PR Checklist

- [x] I've started from master
- [x] I've only committed changes related to this PR
- [x] All Unit tests pass
- [x] I've removed all commented code
- [x] I've removed all unneeded console.log statements

### CODE REVIEW

**Comment by WDaan** [comment]
This is neat! Why didn't I think of this?!

Any way we could do some JSON formatting/validation on pasting?? Might be a bit easier to understand/debug when importing settings

**Comment by Larsluph** [comment]
I looked into it this afternoon, I only found something to validate that it's a JSON but not to validate a certain JSON structure, no idea if it can help to migrate this component to TS 🤔

File: src/components/Settings/Tabs/VueTorrent/VGeneral.vue [hunk 1]
```diff
@@ -242,6 +252,20 @@ export default {
     async fetchQbitVersion() {
       this.Qbitversion = await Qbit.getAppVersion()
     },
+    importSettings() {
+      try {
+        const userState = JSON.parse(this.settingsField)
+      } catch (e) {
+        alert('Invalid JSON!')
```
Comment (WDaan) [inline src/components/Settings/Tabs/VueTorrent/VGeneral.vue:259]: Would probably be better to replace this with something like 
`Vue.$toast.error(....)`

**Comment by WDaan** [comment]
About the validation... I think we might be able to check it against something like `Partial<StoreState>` to enforce the schema... Or maybe use something like JsonSchema.
I'm not yet sure what the best approach would be for something like this.

**Comment by Larsluph** [comment]
Yeah but about that I saw that TS is only check types at compile time, at runtime it's just plain JS so it can't type check in our case

**Comment by WDaan** [comment]
Then perhaps something like JsonSchema, we could generate it from the Typescript type and validate against it. It would probably give some better errors about what's wrong with the provided settings upon invalid configuration.

**Comment by Larsluph** [comment]
Just dropping this here to find it back easily
https://www.npmjs.com/package/jsonschema

**Comment by Larsluph** [comment]
Okay so jsonschema works when I'm using it in a single file, while importing it using `require('jsonschema').validate`.

Problem is that when using `import` statements, the `validate` function doesn't work and throws an error: `resolve is not a function`
https://github.com/tdegrunt/jsonschema/blob/5e219db670bb8a57f5af932e21d51e2ccf44e679/lib/validator.js#L118

**Comment by Larsluph** [comment]
I think I left the json schema check that fails in the PR, the feature might not be usable in the current state.

**Pull request merged**